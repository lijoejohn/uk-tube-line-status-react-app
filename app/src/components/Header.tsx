import { DOM_ELEMENT_IDS, LABELS } from '../constans';

function Header(): JSX.Element {
  return (
    <header className="govuk-header " role="banner" data-module="govuk-header">
      <div className="govuk-header__container">
        <div className="govuk-header__logo govuk-!-margin-left-9 govuk-!-margin-right-9">
          <a href="/" className="govuk-header__link govuk-header__link--homepage">
            <span className="govuk-header__logotype">
              <span
                data-testid={DOM_ELEMENT_IDS.header_link_text}
                id={DOM_ELEMENT_IDS.header_link_text}
                className="govuk-header__logotype-text">
                {LABELS.APP_TITLE}
              </span>
            </span>
          </a>
        </div>
      </div>
    </header>
  );
}
export default Header;
