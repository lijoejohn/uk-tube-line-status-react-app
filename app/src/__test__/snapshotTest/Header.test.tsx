import * as renderer from 'react-test-renderer';
import Header from '../../components/Header';

describe('Header Component', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
