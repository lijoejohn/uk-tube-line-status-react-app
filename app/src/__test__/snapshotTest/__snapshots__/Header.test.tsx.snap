// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
<header
  className="govuk-header "
  data-module="govuk-header"
  role="banner"
>
  <div
    className="govuk-header__container"
  >
    <div
      className="govuk-header__logo govuk-!-margin-left-9 govuk-!-margin-right-9"
    >
      <a
        className="govuk-header__link govuk-header__link--homepage"
        href="/"
      >
        <span
          className="govuk-header__logotype"
        >
          <span
            className="govuk-header__logotype-text"
            data-testid="header_link_text"
            id="header_link_text"
          >
            Tube Status
          </span>
        </span>
      </a>
    </div>
  </div>
</header>
`;
